nototools/noto_names.py

```diff
@@ -0,0 +1,711 @@
+file namess that follow noto conventions, and generates the corresponding
+name table names.  So it is not useful for non-noto fonts.
+"""
+
+import argparse
+import collections
+import datetime
+import glob
+from os import path
+import re
+import sys
+
+from nototools import cldr_data
+from nototools import noto_fonts
+from nototools import tool_utils
+from nototools import unicode_data
+
+from xml.etree import ElementTree as ET
+
+# Standard values used in Noto fonts.
+
+# Regex values returned in NameTableData must start with ^ and end with $,
+# since lint uses this to understand the value is a regex.
+
+GOOGLE_COPYRIGHT_RE = r'^Copyright 20\d\d Google Inc. All Rights Reserved\.$'
+
+ADOBE_COPYRIGHT_RE = (
+    u"^Copyright \u00a9 2014(?:, 20\d\d)? Adobe Systems Incorporated "
+    u"\(http://www.adobe.com/\)\.$")
+
+NOTO_URL = "http://www.google.com/get/noto/"
+
+SIL_LICENSE = (
+    "This Font Software is licensed under the SIL Open Font License, "
+    "Version 1.1. This Font Software is distributed on an \"AS IS\" "
+    "BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express "
+    "or implied. See the SIL Open Font License for the specific language, "
+    "permissions and limitations governing your use of this Font Software.")
+
+SIL_LICENSE_URL = "http://scripts.sil.org/OFL"
+
+APACHE_LICENSE = "Licensed under the Apache License, Version 2.0"
+
+APACHE_LICENSE_URL = "http://www.apache.org/licenses/LICENSE-2.0"
+
+# default file where we store the family name info
+FAMILY_NAME_INFO_FILE = 'family_name_info.xml'
+
+# Represents what and how we write family names in the name table.
+# If limit_original is true, weights become part of the family name,
+# otherwise the family only has Bold and Regular weights so they
+# remain in the subfamily.
+# if use_preferred is true, there are subfamilies that don't fit into
+# Regular Bold BoldItalic Italic, so generate the preferred names.
+# if use_wws is true, there are subfamilies that don't fit into wws,
+# so generate the wws names.
+FamilyNameInfo = collections.namedtuple(
+    'FamilyNameInfo',
+    'limit_original, use_preferred, use_wws')
+
+# Represents expected name table data for a font.
+# Fields expected to be empty are None.  Fields that are expected
+# to be present but have any value are '-'.
+NameTableData = collections.namedtuple(
+    'NameTableData',
+    'copyright_re, original_family, original_subfamily, unique_id, '
+    'full_name, version_re, postscript_name, trademark, manufacturer, '
+    'designer, description_re, vendor_url, designer_url, license_text, '
+    'license_url, preferred_family, preferred_subfamily, wws_family, '
+    'wws_subfamily')
+
+_SCRIPT_KEY_TO_FONT_NAME = {
+    'Aran': 'Urdu',
+    'HST': 'Historic',
+    'LGC': None,
+    'Zsye': None,
+}
+
+
+# copied from noto_lint, we should have a better place for it.
+def preferred_script_name(script_key):
+  try:
+    return unicode_data.human_readable_script_name(script_key)
+  except:
```

Comment:
Could this catch a more specific exception? There's another `except:` later in the file also.